src/render/posts/does-tdd-work.html.md

```diff
@@ -0,0 +1,228 @@
+|[Maximilien](http://faculty.salisbury.edu/~xswang/Research/Papers/SERelated/TDD/p564-maximilien.pdf)|CS|1|9|50% reduction in defect density|minimal impact|
+|[Williams](http://ieeexplore.ieee.org/document/1251029/)|CS|1|9|40% reduction in defect density|no change|
+
+#### In Academia
+|Study|Type|No. of Programmers|Quality Effects|Productivity Effects|
+|-----|----|------------------|---------------|--------------------|
+|[Edwards](http://www.cs.tufts.edu/~nr/cs257/archive/stephen-edwards/automated-feedback.pdf)|CE|59|54% fewer defects|n/a|
+|[Kaufmann](http://dl.acm.org/citation.cfm?id=949421)|CE|8|improved information flow|50% improvement|
+|[Muller](http://ieeexplore.ieee.org/document/1049202/)|CE|19|no change, but better reuse|no change|
+|[Pancur](http://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=1248175)|CE|38|no change|no change|
+|[Erdogmus](http://ieeexplore.ieee.org/document/1423994/)|CE|35|no change|improved productivity|
+
+The data here suggest mostly positive or neutral results from TDD. There is one exception (the previously mentioned George paper) where TDD took 16% longer than the control group but it is noted the control group wrote lest tests in that study.
+
+> The essence of TDD as a design methodology is virtually unstudied, yet scattered early adoption has proceeded based solely on anecdotal evidence.   
+
+> While empirical studies will rarely produce absolute repeatable results, such studies can provide evidence of causal relationships, implying results that will most likely occur in given contexts.  
+
+This to me is a great point. Measurement is difficult and getting fully conclusive results for something like this is almost impossible but we can still use the evidence to help make decisions.
+
+> Survey data reveals that developer opinions about the TDD process improve with TDD experience whereas opinions of test-last programming decrease.  
+
+Much of the data in that paper relies on surveys. These surveys indicated that the more knowledge and experience a developer had with testing and software development in general, the more likely the developer would be of having a better experience with TDD as a process. Interesting but not a major revelation of course. 
+
+Some conclusion the paper arrived at:
+
+>This research has demonstrated that TDD can and is likely to improve some software
+quality aspects at minimal cost over a comparable test-last approach. In particular
+it has shown statistically significant differences in the areas of code complexity,
+size, and testing.
+
+This did appear to be the conclusion based on the data but this is a while back now so it's difficult to say if this is as true today. Also the experiments were on such small groups that it's hard to take this as a fact. To say the results are statistically significant could be a bit of a stretch but it's certainly better than no data.
```

Comment:
I don't understand why the passage of time would have an effect on its relevance?

This is a bit pernickity, but when you say "it's certainly better than no data": this may be true, but as a skeptical reader I'm not inclined to accept that at face value. If I were you, I would either soften the claim or justify it somehow.